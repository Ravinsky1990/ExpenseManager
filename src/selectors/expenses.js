import moment from "moment";

//Visible expenses

export default (expenses, {text, sortBy, startDate, endDate})=>{
    return expenses.filter((expense)=>{
        const createAtmoment = moment(expense.createAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(createAtmoment, "day") : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createAtmoment, "day") : true; 
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase())

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b)=>{
        if (sortBy === "date"){
            return a.createAt < b.createAt ? 1 : -1;
        } else if(sortBy ==="amount"){
            return a.amount < b.amount ? 1 : -1;
        }
    });
}
