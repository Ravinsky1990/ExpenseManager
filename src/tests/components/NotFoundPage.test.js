import React from "react";
import {shallow} from "enzyme";
import NotFoundPage from "../../components/NotFoundPage";
import expenses from "../fixtures/expenses";

test("should render NotfoundPage correctly", ()=>{
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper).toMatchSnapshot()
})