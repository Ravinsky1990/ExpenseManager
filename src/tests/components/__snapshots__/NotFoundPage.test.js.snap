// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotfoundPage correctly 1`] = `
<div>
  404!!!
  <Link
    replace={false}
    to="/"
  >
    Go home
  </Link>
</div>
`;
